awslabs/eventbridge-kafka-connector
# Support custom `TimeMapper` class

TimeMapper interface for setting AWS Event time

Description
-----------

Before this change the time field of the PutEventRequest was always null in which case AWS EventBridge will set it to current timestamp. This PR provides a TimeMapper interface and an optional parameter for overriding the default implementation class name. TimeMapper maps SinkRecord to Instant.

Also made Json operations more available by moving methods from DefaultEventBridgeMapper to public methods in JsonMapper.

-----------



Test Steps
-----------

Added unit tests that verify that 1. the time field is still null with no TimeMapper override 2. with TimeMapper override the time value is changed.


Checklist:
----------




Related Issue
-----------

#358 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
